setup.py

```diff
@@ -321,6 +328,8 @@ def cmake(install_prefix: str = "./nvfuser"):
         cmd_str.append("-DBUILD_NVFUSER_BENCHMARK=ON")
     if BUILD_WITH_ASAN:
         cmd_str.append("-DNVFUSER_BUILD_WITH_ASAN=ON")
+    if BUILD_WITHOUT_DISTRIBUTED:
+        cmd_str.append("-DNVFUSER_DISTRIBUTED=OFF")
```

Comment:
I'm wondering how this would interact with the envvar in our CMakeLists.txt. i.e. what if the two disagrees against each other?

Also, looks like via setuptools, we can only force disable the flag. We still need to go through environment variable to mark it as `enabled`. This feels like something that worth putting down in build doc.

https://github.com/NVIDIA/Fuser/wiki/Developer-guide

Reply:
The dependent option will not let two flags disagree with each other.
If USE_DISTRIBUTED is off then NVFUSER_DISTRIBUTED is forced to off and cannot be changed by the user.
If USE_DISTRIBUTED is on then NVFUSER_DISTRIBUTED is an optional flag that defaults to on. Using`--no-distributed` will set NVFUSER_DISTRIBUTED to off.

If we properly parse USE_DISTRIBUTED from pytorch, then we shouldn't run into any problems.

Reply:
Got'ya. I was a bit nervous earlier with our usage of env var `USE_DISTRIBUTED`. Looks like we are query the proper torch config. :bow: